packages/core/src/browser/resource-context-key.ts

```diff
@@ -29,23 +31,28 @@ export class ResourceContextKey {
     @inject(ContextKeyService)
     protected readonly contextKeyService: ContextKeyService;
 
+    @inject(ApplicationServer)
+    protected readonly applicationService: ApplicationServer;
+
     protected resource: ContextKey<Uri>;
     protected resourceSchemeKey: ContextKey<string>;
     protected resourceFileName: ContextKey<string>;
     protected resourceExtname: ContextKey<string>;
     protected resourceLangId: ContextKey<string>;
     protected resourceDirName: ContextKey<string>;
     protected resourcePath: ContextKey<string>;
+    protected isPosix: boolean;
 
     @postConstruct()
-    protected init(): void {
+    protected async init(): Promise<void> {
         this.resource = this.contextKeyService.createKey<Uri>('resource', undefined);
         this.resourceSchemeKey = this.contextKeyService.createKey<string>('resourceScheme', undefined);
         this.resourceFileName = this.contextKeyService.createKey<string>('resourceFilename', undefined);
         this.resourceExtname = this.contextKeyService.createKey<string>('resourceExtname', undefined);
         this.resourceLangId = this.contextKeyService.createKey<string>('resourceLangId', undefined);
         this.resourceDirName = this.contextKeyService.createKey<string>('resourceDirName', undefined);
         this.resourcePath = this.contextKeyService.createKey<string>('resourcePath', undefined);
+        this.isPosix = await this.applicationService.getBackendOS() !== OS.Type.Windows;
```

Comment:
@msujew, it looks like you could import `isWindows` from `core/src/common/os.ts` rather than relying on the ApplicationServer. That would give you what you need synchronously rather than waiting on a `Promise`.

Reply:
Except `isWindows` will give the OS of the running browser here, and not the remote backend? Not sure which one makes more sense here, but they aren't the same.

Reply:
@colin-grant-work Exactly as @paul-marechal is saying, in this case we need the backend OS, since it specifies the string format of all file paths (forward- vs. backward-slashes, drive letters in front, etc.). The whole reason why this bug exists, is that the frontend and backend might convert the same URI into different strings, which would then result in a mismatch of the `when` context property of commands.

In particular, when we have something like `resourcePath in ext:pkgJsonWithTestScript` as a context, the monaco frontend on windows will convert the current `resourcePath` into something like `\user\settings.json`, while the posix backend path is actually using `/user/settings.json`. Monaco will then perform an `in` check on the `ext:pkgJsonWithTestScript` object, which will fail.

Reply:
I suppose then I wonder whether we shouldn't make retrieving information about the backend OS part of the pre-load startup routine so that we can then access it synchronously from inside the application? That would alleviate some of my anxiety about race conditions.